go/libraries/doltcore/doltdb/table.go

```diff
@@ -34,43 +32,30 @@ import (
 	"github.com/dolthub/dolt/go/store/types"
 )
 
-const (
-	// TableNameRegexStr is the regular expression that valid tables must match.
-	TableNameRegexStr = `^[a-zA-Z]{1}$|^[a-zA-Z_]+[-_0-9a-zA-Z]*[0-9a-zA-Z]+$`
-	// ForeignKeyNameRegexStr is the regular expression that valid foreign keys must match.
-	// From the unquoted identifiers: https://dev.mysql.com/doc/refman/8.0/en/identifiers.html
-	// We also allow the '-' character from quoted identifiers.
-	ForeignKeyNameRegexStr = `^[-$_0-9a-zA-Z]+$`
-	// IndexNameRegexStr is the regular expression that valid indexes must match.
-	// From the unquoted identifiers: https://dev.mysql.com/doc/refman/8.0/en/identifiers.html
-	// We also allow the '-' character from quoted identifiers.
-	IndexNameRegexStr = "^[-`$_0-9a-zA-Z]+$"
-)
-
-var (
-	tableNameRegex      = regexp.MustCompile(TableNameRegexStr)
-	foreignKeyNameRegex = regexp.MustCompile(ForeignKeyNameRegexStr)
-	indexNameRegex      = regexp.MustCompile(IndexNameRegexStr)
-
-	ErrNoConflictsResolved = errors.New("no conflicts resolved")
-)
+var ErrNoConflictsResolved = errors.New("no conflicts resolved")
 
 // IsValidTableName returns true if the name matches the regular expression TableNameRegexStr.
-// Table names must be composed of 1 or more letters and non-initial numerals, as well as the characters _ and -
+// Table names can't end with space characters
 func IsValidTableName(name string) bool {
-	// Ignore all leading digits
-	name = strings.TrimLeftFunc(name, unicode.IsDigit)
-	return tableNameRegex.MatchString(name)
-}
-
-// IsValidForeignKeyName returns true if the name matches the regular expression ForeignKeyNameRegexStr.
-func IsValidForeignKeyName(name string) bool {
-	return foreignKeyNameRegex.MatchString(name)
+	if len(name) == 0 || unicode.IsSpace(rune(name[len(name)-1])) {
+		return false
+	}
+	return IsValidIdentifier(name)
 }
 
-// IsValidIndexName returns true if the name matches the regular expression IndexNameRegexStr.
-func IsValidIndexName(name string) bool {
-	return indexNameRegex.MatchString(name)
+// IsValidIdentifier returns true according to MySQL's quoted identifier rules.
+// Docs here: https://dev.mysql.com/doc/refman/8.0/en/identifiers.html
+func IsValidIdentifier(name string) bool {
```

Comment:
I initially wanted to implement this using a regex, but it appears that the specific syntax needed is not supported
Ref: https://github.com/google/re2/wiki/Syntax

I tried
```
^[\u0001-\uFFFF]+   // doesn't compile
^[\%u0001-\%uFFFF]+ // doesn't compile
^[\%u0001-FFFF]+    // does compile but wrong
```